src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/ReflectionJsonTypeInfoOfT.cs

```diff
@@ -40,7 +40,7 @@ internal ReflectionJsonTypeInfo(JsonConverter converter, JsonSerializerOptions o
                 AddPropertiesAndParametersUsingReflection();
             }
 
-            UntypedCreateObject = Options.MemberAccessorStrategy.CreateConstructor(typeof(T));
+            ((JsonTypeInfo)this).CreateObject = Options.MemberAccessorStrategy.CreateConstructor(typeof(T));
```

Comment:
This is a bit ugly, perhaps we should just modify CreateConstructor - I think it should be one line change